AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/java/org/androidannotations/test/menu/InjectMenuActivityTest.java

```diff
@@ -51,14 +58,34 @@ public void testMenuInjectedFromOnCreateOptionsMenu() {
 	@Test
 	public void methodInjectedMenu() {
 		Menu menu = mock(Menu.class);
+
+		assertThat(injectMenuActivity.menuIsInflated).isFalse();
 		injectMenuActivity.onCreateOptionsMenu(menu);
+		assertThat(injectMenuActivity.menuIsInflated).isTrue();
+
 		assertThat(injectMenuActivity.methodInjectedMenu).isSameAs(menu);
 	}
 
 	@Test
 	public void multiInjectedMenu() {
 		Menu menu = mock(Menu.class);
+
+		assertThat(injectMenuActivity.menuIsInflated).isFalse();
 		injectMenuActivity.onCreateOptionsMenu(menu);
+		assertThat(injectMenuActivity.menuIsInflated).isTrue();
+
 		assertThat(injectMenuActivity.multiInjectedMenu).isSameAs(menu);
 	}
+
+	private InjectMenuActivity.MockMenuInflater createMenuInflater() {
+		final InjectMenuActivity.MockMenuInflater menuInflater = Mockito.mock(InjectMenuActivity.MockMenuInflater.class);
+		doAnswer(new Answer<Void>() {
+			public Void answer(InvocationOnMock invocation) {
+				menuInflater.menuInflated = true;
```

Comment:
We then would need to call `verify()` in the InjectMenuActivity. after the call to `injectMenuActivity.onCreateOptionsMenu(menu);` the inflater is awlays called. the interesting part is that it has to be called BEFORE a method with `@InjectMenu` is called.